packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -48,6 +48,31 @@ export interface ParametersGatherer<T> {
   gather(): Promise<CancelResponse | ContinueResponse<T>>;
 }
 
+export class CompositeParametersGatherer<T> implements ParametersGatherer<T> {
+  protected readonly gatherers: ParametersGatherer<any>[];
```

Comment:
Let's make this one `private` for now. There is no `final` keyword in TypeScript but I don't want anyone to be able to subclass this for now.